src/fmu/dataio/_metadata.py

```diff
@@ -261,24 +265,84 @@ def _populate_meta_masterdata(self) -> None:
     def _populate_meta_access(self) -> None:
         """Populate metadata overall from access section in config + allowed keys.
 
-        Access should be possible to change per object, based on user input.
-        This is done through the access_ssdl input argument.
+        The access block contains the following keys:
+        access:
+          asset: str
+          ssdl:
+            rep_include: [True/False]
+            access_level: ["internal"/"restricted"] # to be deprecated
+          classification: ["internal"/"restricted"]
 
         The "asset" field shall come from the config. This is static information.
+        The "classification" and "rep_include" fields are dynamic. They can be set via
+        input arguments. If not set, they will fall back to defaults from config.
 
-        The "ssdl" field can come from the config, or be explicitly given through
-        the "access_ssdl" input argument. If the access_ssdl input argument is present,
-        its contents shall take presedence.
-
+        WIP: We are deprecating the 'access_ssdl' input argument, replacing it with the
+        'rep_include' and 'classification' arguments. While 'access_ssdl' is deprecated,
+        we still support it. Therefore, we must account for all possible weird
+        combinations for a while.
         """
+
```

Comment:
Building the `access` block a bit more explicitly, then sending it to Pydantic - rather than modifying the config in multiple steps here and there. This also allows for keeping the config and the input argument separate until this point. NOT doing this in `dataio.py` since then we have to deal with the possible repeating steps of checking input arguments.